src/uct/api/uct.h

```diff
@@ -337,7 +337,8 @@ void uct_release_pd_resource_list(uct_pd_resource_desc_t *resources);
  *
  * @return Error code.
  */
-ucs_status_t uct_pd_open(const char *pd_name, uct_pd_h *pd_p);
```

Comment:
config is filled in the call to uct_pd_config_read() which should be called before uct_pd_open(). if this is done then config isn't null (but can be empty).
if you call uct_pd_open with null in config you wont be able to read anything from it.
but according to the api, you do need to call uct_pd_config_read() first.